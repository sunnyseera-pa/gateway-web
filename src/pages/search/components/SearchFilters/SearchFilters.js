import React from 'react';
import { ReactComponent as CDStar } from '../../../../images/cd-star.svg';
import googleAnalytics from '../../../../tracking';

const SearchFilters = ({ filters, onAdvancedSearchClick }) => (
	<>
		{filters && <div className='saved-filterHolder'>{filters}</div>}
		<div className='advanced-search-link-container'>
			<CDStar fill='#f98e2b' height='20' width='20' />
			<a
				href='javascript:void(0)'
				className='textUnderline gray800-14 cursorPointer'
				onClick={() => {
					googleAnalytics.recordEvent('Datasets', 'User clicked advanced search link', 'Advanced search modal opened');
					if (onAdvancedSearchClick) onAdvancedSearchClick();
				}}>
				Advanced Search
			</a>
		</div>
	</>
);

export default SearchFilters;
