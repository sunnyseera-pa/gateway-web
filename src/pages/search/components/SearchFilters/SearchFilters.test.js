import React from 'react';
import SearchFilters from '.';
import { render } from '@testing-library/react';
import googleAnalytics from '../../../../tracking';

let wrapper;

const props = {
	onAdvancedSearchClick: jest.fn(),
};

jest.mock('../../../../tracking');

describe('Given the SearchFilters component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			wrapper = render(<SearchFilters {...props} />);
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		describe('And the data wizard link is clicked', () => {
			beforeAll(() => {
				fireEvent.click(wrapper.getByText('Advanced Search'));
			});

			it('Then calls onClick with step', () => {
				expect(props.onAdvancedSearchClick).toHaveBeenCalled();
			});

			it('Then records an event', () => {
				expect(googleAnalytics.recordEvent).toHaveBeenCalledWith(
					'Datasets',
					'User clicked advanced search link',
					'Advanced search modal opened'
				);
			});
		});

		describe('And filters are added', () => {
			it('Then calls onClick with step', () => {
				wrapper.rerender(<SearchFilters {...props} filters='Filters go here...' />);

				expect(wrapper.getByText('Filters go here...')).toBeTruthy();
			});
		});
	});
});
