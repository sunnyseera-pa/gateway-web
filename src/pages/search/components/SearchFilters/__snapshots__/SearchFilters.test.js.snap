// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the SearchFilters component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="advanced-search-link-container"
  >
    <iconmock
      fill="#f98e2b"
      height="20"
      width="20"
    />
    <a
      class="textUnderline gray800-14 cursorPointer"
      href="javascript:void(0)"
    >
      Advanced Search
    </a>
  </div>
</div>
`;
